test/frontend.js

```diff
@@ -24,35 +26,94 @@ describe('Frontend', function () {
     })
   })
-  /*
-  describe('EJS test page', function () {
-    it('TODO list should have at least two items', function () {
-      // return n.click('nav.nav .nav-center > .nav-item:last-child')
-      return n.click('#testEJS')
-      .wait(() => typeof $ !== 'undefined')
-      .evaluate(() => $('#todo').children().length)
-      .should.finally.greaterThan(1)
+  describe('Test Localization Gathering Function', function () {
+    it('Verify output of transform functions', function (done) {
+      let script = `
+        <template>
+            <i18n comment = "Amazing Test">A test of sorts</i18n>
+            <i18n comment="Amazing Test2">A test of wit</i18n>
+            <i18n>A test of strength</i18n>
+        </template>
+        <script>
+            L('this is some translatable Text','this is relevant commentary')
+            L('this text lacks a comment')
+        </script>
+        
+         `
+      let path = 'script.vue'
+      fs.writeFile(path, script, (err) => {
+        if (err) {
+          throw err
+        }
+        let output = 'translation.json'
+        let args = ['scripts/i18n.js', path, output]
+        exec('node', args, (err, stdout, stderr) => {
+          if (err) {
+            throw err
+          }
+          fs.readFile(output, (err, json) => {
+            if (err) {
+              throw err
+            }
+            let localeObject = JSON.parse(json)
+            should(localeObject).have.property('A test of sorts')
+            should(localeObject).have.property('A test of wit')
+            should(localeObject).have.property('A test of strength')
+            should(localeObject).have.property('this is some translatable Text')
+            should(localeObject).have.property('this text lacks a comment')
+            should(localeObject['A test of sorts']).have.property('comment', 'Amazing Test')
+            should(localeObject['A test of wit']).have.property('comment', 'Amazing Test2')
+            should(localeObject['this is some translatable Text']).have.property('comment', 'this is relevant commentary')
+            should(localeObject['this is some translatable Text']).have.property('text', 'this is some translatable Text')
+            should(localeObject['this text lacks a comment']).have.property('text', 'this text lacks a comment')
+            should(localeObject['A test of sorts']).have.property('text', 'A test of sorts')
+            should(localeObject['A test of wit']).have.property('text', 'A test of wit')
+            should(localeObject['A test of strength']).have.property('text', 'A test of strength')
+            fs.unlink(path, (err) => {
+              if (err) {
+                throw err
+              }
+              fs.unlink(output, (err) => {
+                if (err) {
+                  throw err
+                }
+                done()
+              })
+            })
+          })
+        })
+      })
```

Comment:
This is great. Tip: in the future, if you want to cut down the number of lines here by ~half, you can use `Promise.promisifyAll(require('fs'))` (see their docs) and make this test promise based. That will get rid of most of these closing `})`'s, and you'll be able to get rid of all of the `if (err) throw err` error checking.